core/src/main/java/com/linecorp/armeria/client/DnsCache.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.UnknownHostException;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsRecord;
+
+/**
+ * A cache for DNS responses.
+ */
+@UnstableApi
+public interface DnsCache {
+
+    /**
+     * Returns the default DNS cache.
+     */
+    static DnsCache of() {
```

Comment:
Should we rename this to `ofDefault()`? 🤔

Reply:
I don't know that which one is better between `of()` or `ofDefault()`? 🧐

Reply:
We used `ofDefault()` when we want to emphasize that it's the default behavior, so I guess `ofDefault` should be fine here.

Reply:
Thanks for the explanation that make me clear about the naming.